### PR TITLE
[TensorExpr] Minor cleanup in TensorExprKernel::computeValue

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60041 [TensorExpr] Minor cleanup in TensorExprKernel::computeValue**

Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -2778,30 +2778,18 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
...
     case aten::softmax:
     case aten::log_softmax:
-    case aten::conv2d: {
-      std::vector<ArgValue> argInputs;
-      for (auto inp : inputs) {
-        argInputs.push_back(toArg(inp));
-      }
-      auto outputType = findDtypeForValue(v->node()->output());
-      std::vector<ExprHandle> outputShape = {};
-      // shape inference not implemented for sum
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/kernel.cpp:2788]: > // shape inference not implemented for sum

Do you know why this was done before? Is it okay to remove this check now, as your change is doing?

Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/kernel.cpp:2788]: We're not using shape inference anyway (ie. `aten::sum` is not any special here), the test was just missing `requires_grad=0` in its listing. Previously we did use TE shape inference and that's probably this check was added.

**Review by navahgar — APPROVED** [review]
LGTM

**Review by bertmaher — APPROVED** [review]

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

**Pull request closed without merging**